lib/packages/docutils/rstgen.nim

```diff
@@ -457,8 +457,9 @@ proc generateSymbolIndex(symbols: seq[IndexEntry]): string =
   while i < symbols.len:
     let keyword = symbols[i].keyword
     let cleanedKeyword = keyword.escapeLink
-    result.addf("<dt><a name=\"$2\" href=\"#$2\"><span>$1:</span></a></dt><dd><ul class=\"simple\">\n",
-                [keyword, cleanedKeyword])
+    let prefix = "-" # allows restricting search to match beginning of word, eg: `-split`
+    result.addf("<dt><a name=\"$3\" href=\"#$3\"><span>$1$2:</span></a></dt><dd><ul class=\"simple\">\n",
+                [prefix,keyword, cleanedKeyword])
```

Comment:
1. The diff is ``<span>-$1</span>`` but you managed to make this much harder to see. ;-)
2. Is this an observable change?
3. Can't we do this in a better way? 
4. What problem does this solve?

Reply:
1. done, changed to small diff
2. yes, now top-level entries start with `-`, try this PR with `nim doc -r --project compiler/index` and open index, then enjoy the fact that searching for `-render` brings up only the symbols starting with `render` (as opposed to other symbols with render in the middle, or to random other text (in proc signatures etc) that has `render` in it) => 15 consecutive choices to choose from in your search instead of 79 non-consecutive choices

the `-` blends in well and doesn't look ugly (IANAD(esigner))
3. force users to use a browser plugin that supports regex search to search for `\brender` instead of `-render` or `ode:` instead of `ode\b`
4. allow searching for "just symbols" matching either at beginning (-render) or at end (odule:), without distractors found inside the text (`renderer: renderModule(n: PNode; infile, outfile`):

```
-renderModule: <= the symbol i want to match 
renderer: renderModule(n: PNode; infile, outfile: string; renderFlags: TRenderFlags = {}; fid = FileIndex(-1); conf: ConfigRef = nil)` 
```

Reply:
@araq ping on this